tensorflow/core/platform/env.cc

```diff
@@ -361,6 +374,32 @@ Status WriteStringToFile(Env* env, const string& fname,
   return s;
 }
 
+Status CopyFile(FileSystem* src_fs, const string& src, FileSystem* target_fs,
+                const string& target) {
+  uint64 size;
+  TF_RETURN_IF_ERROR(src_fs->GetFileSize(src, &size));
```

Comment:
From `RandomAccessFile`:

```c
  /// Returns `OUT_OF_RANGE` if fewer than n bytes were stored in `*result`
  /// because of EOF.
```

So you could possibly be able to avoid the round trip of fetching the length, if you check the status returned by `Read`.

Reply:
Thanks. The PR has been updated and size has been removed.